src/wrapper.ts

```diff
@@ -6,109 +6,63 @@
+export const jsHandler = "node_modules/datadog-lambda-js/handler.handler";
+export const jsHandlerFile = "node_modules/datadog-lambda-js/handler.js";
 
-export interface FunctionGroup {
-  funcs: {
-    info: FunctionInfo;
-    method: string;
-  }[];
-  filename: string;
-  runtime: RuntimeType;
-}
-
-export async function writeHandlers(service: Service, funcs: FunctionInfo[], tracingMode: TracingMode) {
-  await cleanupHandlers();
-  await util.promisify(fs.mkdir)(datadogDirectory);
-  const groups = getFunctionGroups(funcs);
-  const promises = groups.map(async (funcGroup) => {
-    const methods = getMethods(funcGroup);
-    const result = getWrapperText(funcGroup.runtime, funcGroup.filename, methods, tracingMode);
-    if (result === undefined) {
+export function redirectHandlers(service: Service, funcs: FunctionInfo[], addLayers: boolean) {
+  const handlerFiles: Set<string> = new Set();
+  funcs.map((func) => {
+    setEnvDatadogHandler(func);
+    const handlerInfo = getDDHandler(func.type, addLayers);
+    if (handlerInfo === undefined) {
       return;
     }
-    const text = result;
-    const filename = await writeWrapperFunction(funcGroup, text);
-    const baseMethodName = path.posix.join(datadogDirectory, funcGroup.funcs[0].info.name);
-    for (const func of funcGroup.funcs) {
-      func.info.handler.handler = `${baseMethodName}.${func.method}`;
-      if (func.info.handler.package === undefined) {
-        func.info.handler.package = {
-          exclude: [],
-          include: [],
-        };
-      }
-      if (func.info.handler.package.include === undefined) {
-        func.info.handler.package.include = [];
-      }
-      func.info.handler.package.include.push(filename);
+    const { handler, handlerFile } = handlerInfo;
+    func.handler.handler = handler;
+
+    if (func.handler.package === undefined) {
+      func.handler.package = {
+        exclude: [],
+        include: [],
+      };
+    }
+    if (func.handler.package.include === undefined) {
+      func.handler.package.include = [];
     }
-    return `${filename}`;
+    func.handler.package.include.push(handlerFile);
+    handlerFiles.add(handlerFile);
   });
-  const files = [...(await Promise.all(promises))];
-  const allFiles = files.filter((file) => file !== undefined) as string[];
-  allFiles.push(path.join(datadogDirectory, "**"));
-  addToExclusionList(service, allFiles);
+  addToExclusionList(service, [...handlerFiles]);
 }
 
-export async function cleanupHandlers() {
-  await removeDirectory(datadogDirectory);
-}
-
-export function getWrapperText(type: RuntimeType, filename: string, methods: string[], tracingMode: TracingMode) {
+function getDDHandler(type: RuntimeType | undefined, addLayers: boolean) {
+  if (type === undefined) {
+    return;
+  }
+  const finalJsHandler = addLayers ? `${jsHandlerLayerPrefix}${jsHandler}` : jsHandler;
+  const finalJsHandlerFile = addLayers ? `${jsHandlerLayerPrefix}${jsHandlerFile}` : jsHandlerFile;
```

Comment:
nit: it might be cleaner to put these within the JS case block so that if the function is Python the definitions get skipped